### ISSUE TITLE
[mdat] Not playable

### ISSUE THREAD

**Issue by rayman3003** [aargirakis/BZRPlayer#192]
Found an audio format that can't be played in BZR2.

BZR's Version: 2.0.21 Alpha

Note: I'm not 100% sure if they r music files or not.

Some samples:

mdat

[mdat.zip](https://github.com/aargirakis/BZRPlayer/files/9768900/mdat.zip)

**Comment by aargirakis** [comment]
You need the corresponding .smp to the .mdat to play those files, then they work.

**Comment by rayman3003** [comment]
> You need the corresponding .smp to the .mdat to play those files, then they work.

OK, I found smpl files in the same folder of mdat files (with same name). But still BZR2 can't play them.

These are all mdat+smpl files that I have. It's not more than 15 samples.

Try them with yourself:

[mdat-smpl.zip](https://github.com/aargirakis/BZRPlayer/files/9776478/mdat-smpl.zip)

**Comment by ciros88** [comment]
confirmed: except for "Chris Hülsbeck - Jim Power (level 4).mdat", the others work with deliplayer

**Comment by aargirakis** [comment]
Seems like the reason for this bug is that tfmx doesn't work with suffix mdat.xxx/smpl.xxx only xxx.mdat/xxx.smpl

**Comment by ciros88** [comment]
i think it is not, i have renamed them as:
![immagine](https://user-images.githubusercontent.com/84340161/219657787-ea77c81a-4051-4c57-8278-6b73bfaa2b18.png)

and no one is working

**Comment by ciros88** [comment]
indeed, any mdat/smpl with prefixes instead of suffixes actually doesnt work in bzr2. support them with prefixes could be a cool feature

**Comment by ciros88** [comment]
indeed this attached mdats are bad/low quality rips: i've compared them with the one from exotica. they are bigger twice than those from exotica (this is why i marked them as low quality or gross rips). i think the burden of beeing able to play this low quality rips isn't from bzr2 but from who made this rips; but..if bzr2 could have a chance to play it, it could be super.

**Comment by aargirakis** [comment]
Actually they play fine if you do TWO things:
Rename to suffix: mdat.xxx/smpl.xxx
Remove ü from the filename

**Comment by ciros88** [comment]
ok so, could u add support for prefixed mdat/smpl (ony if both mdat & smpl files are prefixed or suffixed, not mixed) ?
then #147 will do the rest

**Comment by ciros88** [comment]
> Actually they play fine if you do TWO things: Rename to suffix: mdat.xxx/smpl.xxx Remove ü from the filename

removing the 'ü' char is not needed anymore (thanks to 83dc9d47), while the prefix2suffix renaming is still needed.

what we can do is detecting tfmx at runtime (just relying on the uade detection string content) and, if the filename ends with ".mdat" AND does not start with "mdat." then doing hack it with proper name
